src/rescript/BchRe.res

```diff
@@ -0,0 +1,347 @@
+let defaultRestApi = "https://api.fullstack.cash/v5/"
+type libConfiguration = {
+  restURL: string,
+  apiToken: string,
+  authToken: string,
+}
+@val @scope(("process", "env")) external envRestURL: option<string> = "RESTURL"
+@val @scope(("process", "env")) external envApiToken: string = "BCHJSTOKEN"
+@val @scope(("process", "env")) external envAuthPass: string = "BCHJSAUTHPASS"
+
+type modules =
+  | BitcoinCash
+  | Crypto
+  | Util
+  | Blockchain
+  | Control
+  | Generating
+  | Mining
+  | RawTransactions
+  | Menmonic
+  | Address
+  | HDNode
+  | TransactionBuilder
+  | ECPair
+  | Script
+  | Price
+  | Schnorr
+  | SLP
+  | Encryption
+  | Utxo
+  | Transaction
+  | DSProof
+  | Ecash
+  | Electrumx
+  | PsfSlpIndexer
+
+external requireModule: (string, 'a) => 'a = "require"
+
+let requireInternal = (modules, moduleType) => {
+  switch modules {
+  | BitcoinCash => requireModule("../bitcoincash", moduleType)
+  | Crypto => requireModule("../crypto", moduleType)
+  | Util => requireModule("../util", moduleType)
+  | Blockchain => requireModule("../blockchain", moduleType)
+  | Control => requireModule("../control", moduleType)
+  | Generating => requireModule("../generating", moduleType)
+  | Mining => requireModule("../mining", moduleType)
+  | RawTransactions => requireModule("../raw-transaction", moduleType)
+  | Menmonic => requireModule("../mnemonic", moduleType)
+  | Address => requireModule("../address", moduleType)
+  | HDNode => requireModule("../hdnode", moduleType)
+  | TransactionBuilder => requireModule("../transaction-builder", moduleType)
+  | ECPair => requireModule("../ecpair", moduleType)
+  | Script => requireModule("../script", moduleType)
+  | Price => requireModule("../price", moduleType)
+  | Schnorr => requireModule("../schnorr", moduleType)
+  | SLP => requireModule("../slp/slp", moduleType)
+  | Encryption => requireModule("../ecryption", moduleType)
+  | Utxo => requireModule("../utxo", moduleType)
+  | Transaction => requireModule("../transaction", moduleType)
+  | DSProof => requireModule("../dsproof", moduleType)
+  | Ecash => requireModule("../ecash", moduleType)
+  | Electrumx => requireModule("../electrumx", moduleType)
+  | PsfSlpIndexer => requireModule("../psf-slp-indexer", moduleType)
+  }
+}
+
+module Electrumx = {
+  type t
+  let require = (t) => {
+    requireInternal(Electrumx, t)
+  }
+@new external new: libConfiguration => t = "electrumx"
+}
+
+module RawTransactions = {
+  type t
+external require: string => t = "require"
```

Comment:
You're committing code without running autoformatter. Please make sure you format your code properly.

Reply:
Ok, have format the code and will keep this thing in mind.